OpenRA.Mods.Common/Traits/Passenger.cs

```diff
@@ -115,7 +115,7 @@ bool IsCorrectCargoType(Actor target)
 
 		bool CanEnter(Cargo cargo)
 		{
-			return cargo != null && cargo.HasSpace(Info.Weight);
+			return cargo != null && !cargo.IsTraitDisabled && cargo.HasSpace(Info.Weight);
```

Comment:
this should also be added to ` IsCorrectCargoType`, the one on 102 line. And then perhaps that function also renamed. This will remove the cursor in the same way Carryall or other Cargo cursors are removed.